### PR TITLE
BOAC-37 Distill per-user analytics feed from Canvas stats

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/BOAC-37

Big thanks to @sandeepmjay for the introduction to pandas.

This endpoint will be slow in production, given the number of Canvas requests involved; however, it should provide enough data to construct a demo-worthy view of an individual user.

### CODE REVIEW

File: boac/api/user_controller.py [hunk 1]
```diff
@@ -21,3 +24,39 @@ def user_profile():
         'uid': uid,
         'canvas_profile': canvas_profile,
     })
+
+
+@app.route('/api/user/<uid>/analytics')
+@login_required
+def user_analytics(uid):
+    canvas_profile = canvas.get_user_for_uid(app.canvas_instance, uid)
+    if not canvas_profile:
```
Comment (johncrossman) [inline boac/api/user_controller.py]: This shorthand check is handy... and not immediately intuitive. My education continues.

**Comment by raydavis** [comment]
Fantastic job, @sandeepmjay & @paulkerschen ! I'm eager to see it in action.

**Comment by johncrossman** [comment]
I second @raydavis's approval. Thank you!

File: tests/test_api/test_user_controller.py [hunk 2]
```diff
@@ -34,3 +35,93 @@ def test_includes_canvas_profile_if_available(self, client, fake_auth):
...
+            assert course['analytics']
+
+    def test_course_without_enrollment(self, authenticated_response):
+        """returns a graceful error if the expected enrollment is not found"""
+        course_without_enrollment = TestUserAnalytics.get_course_for_code(authenticated_response, 'BURMESE 1A')
+        assert course_without_enrollment['analytics']['error'] == 'Unable to retrieve analytics'
+
+    def test_course_with_enrollment(self, authenticated_response):
+        """returns sensible data if the expected enrollment is found"""
+        course_with_enrollment = TestUserAnalytics.get_course_for_code(authenticated_response, 'MED ST 205')
```
Comment (raydavis) [inline tests/test_api/test_user_controller.py]: And here's where all the work @paulkerschen put into obfuscating those Canvas fixtures comes into play...

**Pull request merged**